src/client.ts

```diff
@@ -61,6 +61,20 @@ export class Client implements ErdstallClient {
 		return this.erdstallConn.getNftMetadata(token, id, useCache);
 	}
 
+	protected on_internal<T extends EnclaveEvent>(ev: T, cb: ErdstallEventHandler<T>): void {
```

Comment:
You still need to create something like an `internal_watcher` interface with the appropriate methods. THEN you can define the following: `Client.enclaveConn: EnclaveReader & internal_watcher<EnclaveEvent>`. Or however you decided to name your `internal_watcher` interface.

The most important part in my eyes is that the `internal_watcher` methods do not bleed into the public namespace of your `Session` or `Client`.